### PR TITLE
Added Registration

### LINKED ISSUE #25: User should be able to register
Needed for #24.

### PR DESCRIPTION
**Pull request by vaibhavsingh97**
fixes #25 
Screenshot after fix:
image
image
image
image

### CODE REVIEW

**Comment by vaibhavsingh97** [comment]
@tapasweni-pathak @nikhita @jarifibrahim Please review

File: oshc/.idea/inspectionProfiles/profiles_settings.xml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+<component name="InspectionProjectProfileManager">
```
Comment (jarifibrahim) [inline oshc/.idea/inspectionProfiles/profiles_settings.xml]: This file is generated by Pycharm and you shouldn't add it.

Please remove this file and all the files in the `.idea` directory.

File: oshc/main/static/main/css/app.css [hunk 2]
```diff
@@ -1,241 +1,372 @@
...
 * {
-  padding: 0;
-  margin: 0;
-  box-sizing: border-box;
-  -webkit-box-sizing: border-box;
-  -moz-box-sizing: border-box;
-  transition: opacity 0.7s ease-in-out;
-  font-family: "Montserrat", sans-serif
+    padding: 0;
+    margin: 0;
```
Comment (jarifibrahim) [inline oshc/main/static/main/css/app.css]: Most of the changes in this file are related to indentation. Please undo these indentation changes. It's preferable to use a tab size of 2 spaces in `css` files.

File: oshc/main/templates/base.html [hunk 3]
```diff
@@ -2,75 +2,102 @@
...
-				<ul class="nav navbar-nav navbar-right">
-					<li><a href="https://www.youtube.com/channel/UC1_IAby-9me3iICtxuiDL5Q">Sessions</a></li>
-					<li><a href="https://github.com/OpenSourceHelpCommunity">Resources</a></li>
-					<li><a href="mailto:opensourcehelpcommunity@gmail.com">Contact</a></li>
-					<li><a href="https://opensourcehelp.herokuapp.com/">Join Us!</a></li>
-				</ul>
-			</div>
-		</div>
-	</nav>
+<nav class="navbar navbar-inverse navbar-static-top navbar-fixed-top">
```
Comment (jarifibrahim) [inline oshc/main/templates/base.html]: Indentation error.

File: oshc/main/templates/base.html [hunk 4]
```diff
@@ -2,75 +2,102 @@
...
-	<div id="content">
-	{% block content %}{% endblock %}
-	{% block footer %}
-	<footer>
-		<div class="container">
-			<div class="row">
-				<div class="col-xs-12">
-					<div class="social">
-						<ul>
+<div class="container-fluid">
```
Comment (jarifibrahim) [inline oshc/main/templates/base.html]: Why change `content` to `container-fluid`?

File: oshc/main/templates/base.html [hunk 5]
```diff
@@ -2,75 +2,102 @@
...
+    <footer class="footer">
+        <div class="container">
+            <div class="row">
+                <div class="col-xs-12">
+                    <div class="social">
+                        <ul>
 							<span class="col-sm-6 left-align">
 								<li>
-									<a href="https://www.facebook.com/opensourcehelpcommunity/"><i class="fa fa-facebook" aria-hidden="true"></i></a>
+									<a href="https://www.facebook.com/opensourcehelpcommunity/"><i
```
Comment (jarifibrahim) [inline oshc/main/templates/base.html]: Fix indentation.

File: oshc/main/templates/registration/password_reset_confirm.html [hunk 6]
```diff
@@ -0,0 +1,25 @@
+{% extends "base.html" %}
+{% load i18n %}
+
+{% block content %}
+    <section class="main-section">
+        <div class="container">
+
```
Comment (jarifibrahim) [inline oshc/main/templates/registration/password_reset_confirm.html]: Unnecessary new line.

File: oshc/main/templates/registration/password_reset_confirm.html [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+                <form method="post" action=".">
+                    {% csrf_token %}
+                    {{ form.as_p }}
+
+                    <input type="submit" value="{% trans 'Submit' %}"/>
+                </form>
+
+            {% else %}
+
```
Comment (jarifibrahim) [inline oshc/main/templates/registration/password_reset_confirm.html]: Unnecessary new line.

File: oshc/oshc/settings.py [hunk 8]
```diff
@@ -20,11 +22,10 @@
 SECRET_KEY = 'x1-pogt0-b5owbgq0=ui02-4v)bba!bg&1m8_$)8-&13(907qf'
 
 # SECURITY WARNING: don't run with debug turned on in production!
-DEBUG = os.getenv("DEBUG", False)
+DEBUG = os.getenv("DEBUG", True)
```
Comment (jarifibrahim) [inline oshc/oshc/settings.py]: `DEBUG` should be set to `False`.

File: oshc/oshc/settings.py [hunk 9]
```diff
@@ -105,3 +120,12 @@
 STATICFILES_DIRS = (
     os.path.join(BASE_DIR, 'main/'),
 )
+
+EMAIL_BACKEND = 'django.core.mail.backends.smtp.EmailBackend'
+EMAIL_USE_TLS = True
```
Comment (jarifibrahim) [inline oshc/oshc/settings.py]: Change this and following lines to `os.getEnv('EMAIL_USE_TLS)`. This way it is easier to change the email server and related settings.

File: oshc/main/templates/base.html (hunk 3 above)
Comment (jarifibrahim) [inline oshc/main/templates/base.html]: Indentation error in the entire file.

File: oshc/main/templates/registration/password_reset_confirm.html (hunk 6 above)
Comment (jarifibrahim) [inline oshc/main/templates/registration/password_reset_confirm.html]: Unnecessary new line.

File: oshc/oshc/settings.py (hunk 8 above)
Comment (jarifibrahim) [inline oshc/oshc/settings.py]: `DEBUG` should be set to `False` by default. Please undo your changes.

File: oshc/main/templates/base.html (hunk 4 above)
Comment (jarifibrahim) [inline oshc/main/templates/base.html]: Why change `content` to `container-fluid`?

File: oshc/oshc/urls.py [hunk 10]
```diff
@@ -4,4 +4,5 @@
 urlpatterns = [
     url(r'^admin/', admin.site.urls),
     url(r'', include('main.urls')),
+    url(r'^accounts/', include('registration.backends.simple.urls')),
```
Comment (jarifibrahim) [inline oshc/oshc/urls.py:7]: You're using the [simple backend](https://django-registration-redux.readthedocs.io/en/latest/simple-backend.html) and you've created the files (email related) for the [default backend](https://django-registration-redux.readthedocs.io/en/latest/default-backend.html).  You should include `registration.backends.default.urls`.

**Review by jarifibrahim — CHANGES_REQUESTED** [review]
@vaibhavsingh97 This is a huge pull request. You should never make these many changes in one PR.  Always try to make atomic commits. :)

One more thing you should be careful about is the indentation settings in your IDE. I believe most of the changes in this PR have been made by the auto-format feature of the IDE.

BTW, I pulled your code and ran it locally. The registration and login process doesn't seem to work.
Please fix it. We cannot merge untill it works. 

You just need to make some minor changes then we can merge your code. Good job :)

**Comment by vaibhavsingh97** [comment]
@jarifibrahim Thanks for reviewing. Can you please tell me the error?
Can we move to Slack for some discussion?

**Comment by vaibhavsingh97** [comment]
@jarifibrahim Updated the suggested changes. Please review again

File: oshc/main/static/main/css/app.css [hunk 11]
```diff
@@ -10,9 +10,14 @@
...
 
+html {
+  position: relative;
+  min-height: 100%;
+}
+
 body {
   overflow-x: hidden;
-  padding-top: 50px;
+  margin-bottom: 120px;
```
Comment (jarifibrahim) [inline oshc/main/static/main/css/app.css:20]: This change breaks the layout. You might want to remove it.

File: oshc/main/templates/base.html [hunk 12]
```diff
@@ -52,25 +66,25 @@
 									<a href="https://www.youtube.com/channel/UC1_IAby-9me3iICtxuiDL5Q"><i class="fa fa-youtube-play" aria-hidden="true"></i></a>
 								</li>
 							</span>
-							<span class="col-sm-6 backToTop right-align">
+              <span class="col-sm-6 back-to-top right-align">
```
Comment (jarifibrahim) [inline oshc/main/templates/base.html:69]: Indentation error.

File: oshc/main/static/main/css/app.css [hunk 13]
```diff
@@ -44,17 +49,31 @@ a {
   border-color: #24292e;
 }
 
+.container-fluid {
```
Comment (jarifibrahim) [inline oshc/main/static/main/css/app.css]: Generally, it is a bad idea to apply styles to bootstrap classes. What if `container-fluid` is used somewhere else on the same page? It will automatically get `padding-left: 0px` which is not the intended behavior. You should add your own class/id if you wish to tweak a bootstrap element.

File: oshc/main/templates/registration/registration_form.html [hunk 14]
```diff
@@ -0,0 +1,47 @@
...
+                                    <label class="required" for="id_password2">
+                                        {% trans 'Password confirmation:' %}</label>
+                                    <input class="form-control" id="id_password2" name="password2"
+                                           type="password" required/>
+                                    <span class="helptext">{% trans 'Enter the same password as before, for verification.' %}</span>
+                                </p>
+                                <button type="submit"
+                                        class="btn btn-lg btn-primary btn-block">{% trans 'Submit' %}</button>
+                            </div>
+
```
Comment (jarifibrahim) [inline oshc/main/templates/registration/registration_form.html]: Unnecessary new line.

File: oshc/main/templates/base.html [hunk 15]
```diff
@@ -52,25 +66,25 @@
...
+    </div>
+  </footer>
+  {% endblock %}
 
-	<script src="https://ajax.googleapis.com/ajax/libs/jquery/3.2.1/jquery.min.js"></script>
-	<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script>
-	<script src="{% static "main/js/script.js"%}"></script>
+  <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.2.1/jquery.min.js"></script>
+  <script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script>
+  <script src="{% static " main/js/script.js " %}"></script>
```
Comment (jarifibrahim) [inline oshc/main/templates/base.html]: Change `static " main/js/script.js " ` to `static "main/js/script.js" `

**Review by jarifibrahim — CHANGES_REQUESTED** [review]
The login/registration isn't working. Does it work on your machine?
Also, you might want to squash your commits into a single commit. Ping me on slack if you don't know how to do it.

**Comment by vaibhavsingh97** [comment]
@jarifibrahim
capture1

**Comment by vaibhavsingh97** [comment]
@tapasweni-pathak Please review and merge.
Steps to run this PR:
- `python manage.py migrate`
- `python manage.py collectstatic`
- `python manage.py runserver`

**Comment by nikhita** [comment]
Thanks for the instructions, @vaibhavsingh97! I checked the changes locally and it works perfectly. :raised_hands: 

@jarifibrahim @tapasweni-pathak  Since you reviewed the PR, I'll leave it to you to merge it.

**Review by jarifibrahim — APPROVED** [review]

File: oshc/main/templates/base.html (hunk 15 above)
Comment (jarifibrahim) [reply oshc/main/templates/base.html]: @vaibhavsingh97 Ping

**Comment by jarifibrahim** [comment]
Merging this since we have 2 approvals. Well done @vaibhavsingh97
